pennylane/workflow/qnode.py

```diff
@@ -96,13 +100,96 @@ def _make_execution_config(
 
     return qml.devices.ExecutionConfig(
         interface=getattr(circuit, "interface", None),
-        gradient_method=_gradient_method,
+        gradient_keyword_arguments=gradient_kwargs,
+        gradient_method=diff_method,
         grad_on_execution=grad_on_execution,
         use_device_jacobian_product=execute_kwargs.get("device_vjp", False),
         mcm_config=mcm_config or qml.devices.MCMConfig(),
     )
 
 
+def _update_mcm_config(mcm_config: "qml.devices.MCMConfig", interface: str, finite_shots: bool):
+    """Helper function to update the mid-circuit measurements configuration based on
+    execution parameters"""
+
+    if not finite_shots:
+        mcm_config.postselect_mode = None
+        if mcm_config.mcm_method == "one-shot":
+            raise ValueError(
+                f"Cannot use the '{mcm_config.mcm_method}' method for mid-circuit measurements with analytic mode."
+            )
+
+    if mcm_config.mcm_method == "single-branch-statistics":
+        raise ValueError("Cannot use mcm_method='single-branch-statistics' without qml.qjit.")
+
+    if interface == "jax-jit" and mcm_config.mcm_method == "deferred":
+        # This is a current limitation of defer_measurements. "hw-like" behaviour is
+        # not yet accessible.
+        if mcm_config.postselect_mode == "hw-like":
+            raise ValueError(
+                "Using postselect_mode='hw-like' is not supported with jax-jit when using "
+                "mcm_method='deferred'."
+            )
+        mcm_config.postselect_mode = "fill-shots"
+
+    if (
+        finite_shots
+        and "jax" in interface
+        and mcm_config.mcm_method in (None, "one-shot")
+        and mcm_config.postselect_mode in (None, "hw-like")
+    ):
+        mcm_config.postselect_mode = "pad-invalid-samples"
+
+
+def _resolve_execution_config(
+    tapes: QuantumScriptBatch,
+    device: "qml.devices.Device",
+    execution_config: "qml.devices.ExecutionConfig",
+    transform_program: TransformProgram,
+) -> "qml.devices.ExecutionConfig":
+    """Resolves the execution configuration for non-device specific properties.
+
+    Args:
+        tapes (QuantumScriptBatch): a batch of tapes
+        device (qml.devices.Device): a Pennylane device
+        execution_config (qml.devices.ExecutionConfig): an execution config to be executed on the device
+        transform_program (TransformProgram): a program of transformations to be applied to the tapes
+
+    Returns:
+        qml.devices.ExecutionConfig: resolved execution configuration
+    """
+    updated_values = {}
+    updated_values["gradient_keyword_arguments"] = dict(execution_config.gradient_keyword_arguments)
+
+    if (
+        device.name == "lightning.qubit"

```

Comment:
```suggestion
        "lightning" in device.name
```
Potentially out of scope of this PR, but looking over this I realized the same bug might apply to kokkos and gpu too.